TestJcmd_0

All extended testing.
```
FAILED: testJcmdHelps
java.lang.AssertionError: Help text corrupt: [Usage : jcmd <vmid> <arguments>, ,    -J : supply arguments to the Java VM running jcmd,    -l : list JVM processes on the local machine,    -h : print this help message, ,    <vmid> : Attach API VM ID as shown in jps or other Attach API-based tools, ,    arguments:,       help : print the list of diagnostic commands,       help <command> : print help for the specific command,       <command> [command arguments] : command from the list returned by "help", , list JVM processes on the local machine. Default behavior when no options are specified., , NOTE: this utility may significantly affect the performance of the target JVM.,     The available diagnostic commands are determined by,     the target VM and may vary between VMs.] expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:96)
	at org.testng.Assert.failNotEquals(Assert.java:776)
	at org.testng.Assert.assertTrue(Assert.java:44)
	at org.openj9.test.attachAPI.TestJcmd.testJcmdHelps(TestJcmd.java:87)
```

--- a/jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java
+++ b/jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java
@@ -96,9 +96,10 @@ public class DiagnosticUtils {
 	public static final String DIAGNOSTICS_OPTION_SEPARATOR = ","; //$NON-NLS-1$
 
 	/**
-	 * Report only live heap objects.
+	 * Report live or all heap objects.
 	 */
-	public static final String LIVE_OPTION = "live"; //$NON-NLS-1$
+	private static final String ALL_OPTION = "all"; //$NON-NLS-1$
+	private static final String LIVE_OPTION = "live"; //$NON-NLS-1$
 	private static final String THREAD_LOCKED_SYNCHRONIZERS_OPTION = "-l"; //$NON-NLS-1$
 
 	private static final Map<String, Function<String, DiagnosticProperties>> commandTable;
@@ -169,18 +170,34 @@ public class DiagnosticUtils {
 	}
 
 	private static DiagnosticProperties getHeapStatistics(String diagnosticCommand) {
-		boolean doLive = Arrays.stream(diagnosticCommand.split(DiagnosticUtils.DIAGNOSTICS_OPTION_SEPARATOR))
-				.anyMatch(Predicate.isEqual(DiagnosticUtils.LIVE_OPTION));
-		if (doLive) {
-			runGC();
+		DiagnosticProperties result = null;
+		boolean invalidArg = false;
+		boolean doLive = false;
+		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
+		if (parts.length > 2) {
+			invalidArg = true;
+		} else if (parts.length == 2) {
+			if (LIVE_OPTION.equalsIgnoreCase(parts[1])) {
+				doLive = true;
+			} else if (!ALL_OPTION.equalsIgnoreCase(parts[1])) {
+				invalidArg = true;
+			}
 		}
-		String result = getHeapClassStatisticsImpl();
-		String lineSeparator = System.lineSeparator();
-		final String unixLineSeparator = "\n"; //$NON-NLS-1$
-		if (!unixLineSeparator.equals(lineSeparator)) {
-			result = result.replace(unixLineSeparator, lineSeparator);
+		if (invalidArg) {
+			result = DiagnosticProperties.makeErrorProperties("Command not recognized: " + diagnosticCommand); //$NON-NLS-1$
+		} else {
+			if (doLive) {
+				runGC();
+			}
+			String hcsi = getHeapClassStatisticsImpl();
+			String lineSeparator = System.lineSeparator();
+			final String unixLineSeparator = "\n"; //$NON-NLS-1$
+			if (!unixLineSeparator.equals(lineSeparator)) {
+				hcsi = hcsi.replace(unixLineSeparator, lineSeparator);
+			}
+			result = DiagnosticProperties.makeStringResult(hcsi);
 		}
-		return DiagnosticProperties.makeStringResult(result);
+		return result;
 	}
 
 	private static DiagnosticProperties getThreadInfo(String diagnosticCommand) {
@@ -302,7 +319,9 @@ public class DiagnosticUtils {
 	@SuppressWarnings("nls")
 	private static final String DIAGNOSTICS_GC_CLASS_HISTOGRAM_HELP = "Obtain heap information about a Java process%n"
 			+ FORMAT_PREFIX + DIAGNOSTICS_GC_CLASS_HISTOGRAM + " [options]%n"
-			+ " Options: -all : include all objects, including dead objects%n"
+			+ " Options:%n"
+			+ "          all : include all objects, including dead objects (this is the default option)%n"
+			+ "         live : include all objects after a global GC collection%n"
 			+ "NOTE: this utility may significantly affect the performance of the target VM.%n";
 
 	@SuppressWarnings("nls")

--- a/jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Jmap.java
+++ b/jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Jmap.java
@@ -120,21 +120,37 @@ public class Jmap {
 		int optionsSelected = 0;
 		String errorMessage = null;
 		vmids = new ArrayList<>();
-		for (String a : args) {
-			if (!a.startsWith("-")) {
-				vmids.add(a);
-			} else if (a.startsWith("-histo")) {
-				histo = true;
-				optionsSelected += 1;
-				if (a.endsWith(":live")) {
-					live = true;
-				}
+		for (String arg : args) {
+			if (!arg.startsWith("-")) {
+				vmids.add(arg);
 			} else {
-				errorMessage = "unrecognized option " + a;
-				break;
+				boolean invalidArg = false;
+				String[] parts = arg.split(":");
+				if (parts.length > 2) {
+					invalidArg = true;
+				} else if ("-histo".equalsIgnoreCase(parts[0])) {
+					if (parts.length == 2) {
+						if ("live".equalsIgnoreCase(parts[1])) {
+							live = true;
+						} else {
+							invalidArg = true;
+						}
+					}
+					if (!invalidArg) {
+						histo = true;
+						optionsSelected += 1;
+					}
+				} else {
+					invalidArg = true;
+				}
+				
+				if (invalidArg) {
+					errorMessage = "unrecognized option " + arg;
+					break;
+				}
 			}
 		}
-		if (1 != optionsSelected) {
+		if ((errorMessage == null) && (optionsSelected != 1)) {
 			errorMessage = "exactly one option must be selected";
 		}
 		if (null != errorMessage) {

--- a/test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJcmd.java
+++ b/test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJcmd.java
@@ -82,7 +82,7 @@ public class TestJcmd extends AttachApiTest {
 		for (String helpOption : HELP_OPTIONS) {
 			List<String> jcmdOutput = runCommandAndLogOutput(Collections.singletonList(helpOption));
 			/* Sample of the text from Jcmd.HELPTEXT */
-			String expectedString = "run diagnostic command"; //$NON-NLS-1$
+			String expectedString = "list JVM processes on the local machine."; //$NON-NLS-1$
 			Optional<String> searchResult = StringUtilities.searchSubstring(expectedString, jcmdOutput);
 			assertTrue(searchResult.isPresent(), "Help text corrupt: " + jcmdOutput); //$NON-NLS-1$
 		}
@@ -165,7 +165,7 @@ public class TestJcmd extends AttachApiTest {
 		List<String> args = new ArrayList<>();
 		args.add(getVmId());
 		args.add(GC_CLASS_HISTOGRAM);
-		args.add("-all"); //$NON-NLS-1$
+		args.add("all"); //$NON-NLS-1$
 		List<String> jcmdOutput = runCommandAndLogOutput(args);
 		String expectedString = commandExpectedOutputs.getOrDefault(GC_CLASS_HISTOGRAM, "Test error: expected output not defined"); //$NON-NLS-1$
 		log("Expected string: " + expectedString); //$NON-NLS-1$

